src/Parser/MarksmanshipHunter/Modules/Spells/Trueshot.js

```diff
@@ -0,0 +1,40 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+
+class QuickShot extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  trueShotCDReduction = 0;
+  totalFromCD = 0;
+
+  get tsAvailableCasts() {
+    const player = this.combatants.selected;
+    const quickShotRank = player.traitsBySpellId[SPELLS.QUICK_SHOT_TRAIT.id];
+
+    if (quickShotRank === 0) {
+      return;
+    }
+    else if (quickShotRank > 0 && quickShotRank < 4) {
+      this.trueShotCDReduction = quickShotRank * 10;
+    } else if (quickShotRank === 4) {
+      this.trueShotCDReduction = 38;
+    }
+    else if (quickShotRank === 5) {
+      this.trueShotCDReduction = 45;
+    }
+    else if (quickShotRank === 6) {
+      this.trueShotCDReduction = 52;
+    }
+    else if (quickShotRank === 7) {
+      this.trueShotCDReduction = 58;
+    }
```

Comment:
It's strange the CDR goes first by 10% and then decreases (which messes up the computations) but I suggest something like this instead:
```jsx
if (quickShotRank < 4) {
   this.trueShotCDReduction = quickShotRank * 10;
}
else {
   switch (quickShotRank) {
      case 4:
         this.trueShotCDReduction = 38;
         break;
      case 5:
         this.trueShotCDReduction = 45;
         break;
      case 6:
         this.trueShotCDReduction = 52;
         break;
      case 7:
         this.trueShotCDReduction = 58;
         break;
      default:
         break;
   }
}
```

Reply:
oh yeah, I completely forgot about switch-cases.. derp, way better solution

Reply:
tbh I prefer the if/else 😂